site/content/contribute/gsoc/_index.md

```diff
@@ -0,0 +1,223 @@
+
+-   **Expected outcomes**: Design of the feature: will it be a plugin of its own, or a modification of the Zoom and Jitsi plugin? Implementation of the banner for both cases.
+-   **Skills required**: React
+-   **Difficulty**: Easy
+-   **Mentors**: Daniel Espino García, Alejandro García Montoro
+
+## Apps
+
+Mattermost was designed from the very beginning to be extensible. Up until now, plugins were the main way (along with incoming and outgoing webhooks and slash commands) of integrating third-party services into the product, and while they have been extremely powerful and successful, we are investigating new ways of letting developers extend the functionality.
+
+The result of that investigation is the concept of Apps: the new way to integrate external services into Mattermost.
+
+### App Form Fields
+
+One of the interaction points with Apps are Apps Forms. Apps Forms allow the user to introduce information to send to the App. Right now, we support some basic fields in the forms, like text, bools, or select but we would like to add more complex types and interactions, like date picker, time picker, lists, or checkbox tables.
+
+Apps are designed with mobile support as a first class citizen, so every new form implemented has to support both the web application and the mobile apps.
+
+This project will be developed in close contact with the UX team to better define each individual interaction.
+
+-   **Expected outcomes**: The design and implementation of at least two new form fields. This project can be adapted to the student to implement as many form fields as desired.
+-   **Skills required**: React, React Native
+-   **Difficulty**: Medium
+-   **Mentors**: Daniel Espino García, Alejandro García Montoro
+
+### Apps Locations
+
+One of the ways Mattermost can interact with the new Apps is through the App Locations: Specific locations in the UI (both in the web application and the mobile apps) that can be attached to an action handled by the App. Current locations include buttons in the channel header and the post menu.
+
+The goal of this project is to consider and add other locations, like entries in the main menu, additions to the channel menu or to the configuration menu.
+
+One particularly interesting location to add is one to support audio rooms. There has been a lot of interest in supporting audio rooms in Mattermost, similar to the ones in Discord. There are some proofs of concept, both for Lyno and Mumble, but they are based on hacks to modify the UI of the product. This location would allow an app to host an audio only chat room in Mattermost and display that room on the channel sidebar, displaying the current status of the room.
```

Comment:
```suggestion
One particularly interesting location to consider adding is one to support audio rooms. There has been a lot of interest in supporting audio rooms in Mattermost, similar to the ones in Discord. There are some proofs of concept, both for Lyno and Mumble, but they are based on hacks to modify the UI of the product. This location would allow an app to host an audio-only chat room in Mattermost, and display that room on the channel sidebar, displaying the current status of the room.
```